src/Kafka/ManageKafkaPermissions/components/ManagePermissionsModal.tsx

```diff
@@ -0,0 +1,94 @@
+import {
+  Button,
+  Form,
+  FormGroup,
+  Modal,
+  SelectOptionObject,
+} from "@patternfly/react-core";
+import { useRef } from "react";
+import { useTranslation } from "react-i18next";
+import { SubmitButton } from "./PermissionsModalSubmitButton";
+import { Validated, SelectAccount } from "./SelectAccount";
+
+export enum PrincipalType {
+  UserAccount = "USER_ACCOUNT",
+  ServiceAccount = "SERVICE_ACCOUNT",
+}
+
+export type Principal = {
+  id: string;
+  principalType?: PrincipalType;
+  displayName?: string;
+};
+
+type ManagePermissionsModalProps = {
+  principal: Principal[];
+  hideModal: () => void;
+  kafkaName?: string;
+  selectedAccount: Validated<string | undefined | SelectOptionObject>;
+  onChangeAccount: React.Dispatch<
+    React.SetStateAction<Validated<string | undefined | SelectOptionObject>>
+  >;
+};
+
+export const ManagePermissionsModal: React.FC<ManagePermissionsModalProps> = ({
+  hideModal,
+  kafkaName,
+  principal,
+  selectedAccount,
+  onChangeAccount,
+}) => {
+  const { t } = useTranslation(["manage-kafka-permissions"]);
+  const escapeClosesModal = useRef<boolean>(true);
+  /*value of step will be used when working on step 2
+  const [step, setStep] = useState<number>(1);
+  */
+  const onEscapePress = () => {
+    if (escapeClosesModal.current) {
+      hideModal();
+    }
+  };
+
+  const setEscapeClosesModal = (closes: boolean) => {
+    escapeClosesModal.current = closes;
+  };
+  const ModalForm: React.FunctionComponent = () => (
+    <Form>
+      <FormGroup
+        fieldId="kafka-instance-name"
+        label={t("manage_permissions_dialog.kafka_instance_title")}
+      >
+        {kafkaName}
+      </FormGroup>
+      <SelectAccount
+        id={selectedAccount}
+        onChangeAccount={onChangeAccount}
+        initialOptions={principal}
+        onEscapeModal={setEscapeClosesModal}
+      />
+    </Form>
+  );
+
+  return (
+    <Modal
+      id="manage-permissions-modal"
+      variant={"large"}
+      isOpen={true}
+      aria-label={t("manage_permissions_dialog.aria_label")}
+      position="top"
+      title={t("manage_permissions_dialog.title")}
+      showClose={true}
+      aria-describedby="modal-message"
```

Comment:
The value defined here should reference an `id` that's defined for the primary message of this modal. We don't really have a primary message, but we do state the name of the kafka instance. To make this functional, we should add `id="modal-message"` to the `<FormGroup>` earlier in this document that includes the kafka instance name. This way, when a screen reader announces the modal, it will also provide the kafka instance name for additional context to the user. 

```
      <FormGroup
        fieldId="kafka-instance-name"
        label={t("manage_permissions_dialog.kafka_instance_title")}
        id="modal-message"
      >
        {kafkaName}
      </FormGroup>
```